style(3000): Tune 3000 sizing

Twixes opened the pull request:
## Changes

This tackles a few things that were, in my view, off in 3000:
- navbar spacing was very luxurious, as opposed to our goal for the whole app
- a bunch of places used sizes `rem` values that weren't divisible by 16, such as `0.3rem` or `1.8rem`, which essentially correspond to `3.2px` and `28.8px`. It's safer to use values divisible by 16 or at least 32 so that sizing is more predictable, as otherwise we force the browser to do a bunch of rounding for us
- Insight icons weren't fully polished, as we discussed on Slack
- The homepage cards had a scrollbar in 3000, because of a hidden border making all buttons 2px taller, even though tertiary ones don't have fancy 3D styles and don't need that border
- Also fixed an issue in breadcrumbs where I left some margin in

## How did you test this code?

Take a look at UI snapshot changes.

Twixes committed:
Fix unwanted breadcrumbs margin

Twixes committed:
Reduce navbar spacing

Twixes committed:
Remove breadcrumbs action button sizing hack

Twixes committed:
Round button sizing to full pixels

Twixes committed:
Round table sizing to full pixels

Twixes committed:
Fix homepage cards being scrollable and tune insight icon colors

Twixes committed:
Tune negative margin

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

daibhin commented on frontend/src/layout/navigation-3000/Navigation.scss:
```diff
@@ -71,7 +71,7 @@
 
         li + li {
-            margin-top: 0.25rem;
+            margin-top: 1px;
```
This could become a `space-y-px` class on `Navbar3000__content`

Twixes replied:
Hmm, it could, but then we'd have more of a mix of utility classes and BEM. Gonna stick with BEM here

daibhin commented on frontend/src/lib/components/CompactList/CompactList.tsx:
```diff
@@ -27,9 +27,7 @@ export function CompactList({
-            <div className="spacer-container">
-                <LemonDivider />
-            </div>
+            <LemonDivider className="mx-2 my-2" />
```
```suggestion
            <LemonDivider className="m-2" />
```

daibhin commented on frontend/src/lib/lemon-ui/LemonTable/LemonTable.scss:
```diff
@@ -173,14 +173,14 @@
-                    padding-top: 0.3rem;
-                    padding-bottom: 0.3rem;
+                    padding-top: 0.3125rem;
+                    padding-bottom: 0.3126rem;
```
Just checking if this is a typo?
```suggestion
                    padding-bottom: 0.3125rem;
```

Twixes replied:
That is a typo!

corywatilo reviewed (changes requested):
## Feedback

- [ ]  Nav looks much better, though icon sizes could be shrunk a bit more (`1.5rem` → `1.25rem`) and even less vertical padding. Feels like column should have a min-width though, so the width isn't always dependent on the longest nav item.

- [ ] Would prefer these buttons to be not this tall:

<img width="312" alt="image" src="https://github.com/PostHog/posthog/assets/154479/d9785ecf-c6d8-469f-a594-4b7d2924f463">

Primary button looks "okay" (still too tall), and _definitely_ too tall in the tab buttons. Going for tighter, more condensed in general. (See most of GitHub's buttons for sizing reference.)

- [ ]  Here's how I'd ideally like to see these icons:

    - Smaller
    - Anchored to top
    - (And that funnel icon needs to be shrunk down to match the size of the trend icon but that's an icon package thing I'll handle separately...)

<img width="521" alt="image" src="https://github.com/PostHog/posthog/assets/154479/3c7be6de-c9fd-4d85-a590-03b276c0d971">

Twixes committed:
Update frontend/src/lib/lemon-ui/LemonTable/LemonTable.scss

Co-authored-by: David Newell <d.newell1@outlook.com>

Twixes committed:
Update frontend/src/lib/components/CompactList/CompactList.tsx

Co-authored-by: David Newell <d.newell1@outlook.com>

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

Twixes committed:
Merge branch 'master' into 3000-sizing-fixes

Twixes committed:
Make breadcrumbs buttons small

Twixes committed:
Refactor `> span` to `.LemonButton__chrome`

Twixes committed:
Make navbar spacing tighter

Twixes committed:
Merge branch 'master' into 3000-sizing-fixes

Twixes committed:
Rework project homepage list items

Twixes commented:
@corywatilo 
1. Gonna leave that for later as this is tricky on two ways: 1. We'd have to do even more overrides for the style of buttons in the navbar, which is not great for maintainability. I would love if we were able to use a standard button size defined in the design system there, but if that proves tricky to codify, will do the overrides in a separate PR. There's also the issue that the `ProfilePicture` component currently doesn't have a `1.25rem`-sized variant, it's currently `1.125rem` or `1.5rem` – something to standardize too.
2. Fixed the top bar buttons not being small anymore, that was a regression.
3. Thanks for the design for the project homepage lists. I implemented it, but it turned out that with the context line being the same size as the item name, the buttons would actually have to be taller to maintain their paddings. Ended up making the context smaller:
     <img width="374" alt="Screenshot 2023-11-22 at 11 27 23" src="https://github.com/PostHog/posthog/assets/4550621/6a4e7430-4c9e-436e-af1a-750dba4e80cd">

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

Twixes committed:
Merge branch 'master' into 3000-sizing-fixes

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

Twixes committed:
Fix search icon

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

Twixes commented:
@corywatilo Lemme know about any further tweaks, merged this for now to get some of those fixes in

daibhin commented on frontend/src/layout/navigation-3000/Navigation.scss:
```diff
@@ -77,7 +72,7 @@
 
         li + li {
-            margin-top: 0.25rem;
+            margin-top: -1px;
```
Sorry for the late review here but looks like this is causing nav buttons to overlap
![overlap](https://github.com/PostHog/posthog/assets/6685876/d99f1c1c-3fba-4db4-8232-8a37798fd1a5)
